src/libstd/sys_common/net.rs

```diff
@@ -177,9 +177,20 @@ pub fn lookup_host(host: &str) -> io::Result<LookupHost> {
     };
     let mut res = ptr::null_mut();
     unsafe {
-        cvt_gai(c::getaddrinfo(c_host.as_ptr(), ptr::null(), &hints,
-                               &mut res))?;
-        Ok(LookupHost { original: res, cur: res })
+        match cvt_gai(c::getaddrinfo(c_host.as_ptr(), ptr::null(), &hints, &mut res)) {
+            Ok(_) => {
+                Ok(LookupHost { original: res, cur: res })
+            },
+            Err(e) => {
+                if cfg!(unix) {
+                    // The lookup failure could be caused by using a stale /etc/resolv.conf.
+                    // See https://github.com/rust-lang/rust/issues/41570.
+                    // We therefore force a reload of the nameserver information.
+                    c::res_init();
```

Comment:
This code is in `sys_common` so it is compiled on all platforms, but I don't see a `res_init` on Windows.